2.5.1 backports 1

gsmet opened the pull request:
Please don't merge, I will merge it myself.

zakkak committed:
Back GraalVM.Version implementation with org.graalvm.home.Version

This prevents us from re-inventing the wheel and allows us to compare
beyond major and micro.

(cherry picked from commit 0c55cac428d81249b003231012c763d48a0b1f92)

zakkak committed:
Extend GraalVM.Version to detect JDK version as well

(cherry picked from commit f2fc4b0ff002a951bab963941eb71b38e1780d6f)

zakkak committed:
Work around for https://github.com/quarkusio/quarkus/issues/21372

(cherry picked from commit dc9a6ec5593928ac58681d21de8bc6a8a3de483f)

Commit:
Support "Uncategorized" extensions

(cherry picked from commit afebc6b55670a99db3711e50e7d5eaec1b981440)

stuartwdouglas committed:
Only warn about loggable messages

If messages are dropped that would have been loggable then we display
the warning, otherwise we ignore it as the end result is the same.

(cherry picked from commit f5c8dcbcc572ece47c2daf4e7475ecfcb5adff58)

michalszynkiewicz committed:
Don't fail native test start when http is not exposed

fixes #21317

(cherry picked from commit 7e43de290aa50fad4cdd5925b38a6d4eae8bf9bd)

sberyozkin committed:
Update the CONTRIBUTING Code Review section

(cherry picked from commit 53681dc0b2bea18b206f29de347b4d49f0f0530f)

sberyozkin committed:
Update OIDC Dev UI to support extra grant options

(cherry picked from commit 24c315993ab10073950cebd0df7bc26f6641ac1f)

radcortez committed:
Ignore ConfigMapping injection points annotations for type matching

(cherry picked from commit 060a1b25ccdc99801b4b08c4eea4269bb5e81559)

ozangunalp committed:
Confluent kafka avro serializer register NullContextNameStrategy for reflection

Fixes #21412

(cherry picked from commit ac31451f8a67443471c828330cfbb2bbe0e9c701)

dependabot[bot] committed:
Bump com.gradle.plugin-publish from 0.17.0 to 0.18.0 in /devtools/gradle

Bumps com.gradle.plugin-publish from 0.17.0 to 0.18.0.

---
updated-dependencies:
- dependency-name: com.gradle.plugin-publish
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
(cherry picked from commit 64702b56e8c67a5097bf2eca1b527d1d0ffb43f8)

famod committed:
Don't build docs module in CI JVM Tests

(cherry picked from commit 8fbe9ba971eb44b043cbac511480412f844b6948)

shawkins committed:
fix to use serviceAccountName, instead of serviceAccount

(cherry picked from commit 80d8c2e2250c017d34033b1947b274b497ec878d)

ia3andy committed:
Make uncategorized work for extensions without categories

(cherry picked from commit fa9c9027a9a839814371a9cdaefcd42bad8dbab8)

gastaldi committed:
Include the logger names (and levels) in the message

See https://github.com/quarkusio/quarkus/pull/21473#discussion_r750007834

(cherry picked from commit 14aa776bc38ebee286be649a9734e1b16e45c07a)

famod committed:
Pass on compiler release flag value in Maven dev mode

(cherry picked from commit 4743568614f8f302cd121934430d053e79f3a642)

xstefank committed:
Health-UI calls the wrong URL for health in dev mode

(cherry picked from commit b1f303248aff29fdec5e0fa9a9ee8f9c54be22d8)

Sanne committed:
Fixing failing test in Hibernate Reactive: SchemaUpdateTest

(cherry picked from commit a303920041369123888db25e73121cb2125d145f)

phillip-kruger committed:
OpenAPI: Do not remove headers added by client when adding default CORS Headers
Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

(cherry picked from commit f2a2a783c24f60c98e33fce10fa19ba41ff37296)

famod committed:
Fix mongodb-client health test after client creation was made lazy

(cherry picked from commit 4c542eb709073b7d908c803a7349feb0db0d6037)

Sanne committed:
Fix Kerberos module use in Java 17

(cherry picked from commit ed6339ef1706116a9211a35467553ca09e030e6d)

ia3andy committed:
Fix problem with dependency order

(cherry picked from commit 0fff998bb5278a0da60065df9773bc8eff1450d4)

carlesarnal committed:
Upgrade apicurio version

(cherry picked from commit 4be4a8d3e8dc9fac02b1da1b1fb52aa2cb0607c2)

galderz committed:
Push `sun.lwawt.macos` package to runtime init in Darwin in Java 17

(cherry picked from commit 918d4c36be669b8c110c24490b73a041436e77d8)

gemmellr committed:
Update JMS guide to not create generated [test] code that isnt used, again.

Missed fix in earlier change in #21364

(cherry picked from commit 331559a104a059c4b0f39b56f3ba45636c3baeba)

sberyozkin committed:
Add JWT and POST body tests

(cherry picked from commit 09378b759feffd78f8441c5b02b069968ca08153)

stuartwdouglas committed:
Fix issue with DevMode tests

Also fix issue where close() might not be called on startup failure in
dev mode tests, potentially causing unrelated tests to also fail.

(cherry picked from commit a325175e474d8e7dcc9a7f2c279c412cbb1c2e14)

geoand committed:
Change link for testcontainers usage example

Fixes: #21508
(cherry picked from commit 0bc6134f7c76e3ad59e77a02fb2f29b8d286a98c)

knutwannheden committed:
Fix CME in MicrometerRecorder shutdown hook

The collection returned by `Metrics.globalRegistry.getRegistries()` is not a copy, so later calls to `Metrics.removeRegistry()` end up modifying the same underlying collection and thus results in `ConcurrentModificationException`s in the for loop of the shutdown hook registered by `MicrometerRecorder#configureRegistries()`.

Fixes #21604

(cherry picked from commit 9acba2106c9c27be102dddaf7173bdc10a816c3b)

geoand committed:
Don't warn when QuarkusTestResourceLifecycleManager can't find field

Fixes: #21577
(cherry picked from commit ff1a50a3844eb6ba32b6cdfaea92cbbd83363bb6)

TomasHofman committed:
Fix RestClientFallbackConfigSourceInterceptor#iterateNames()

The fallback interceptor shouldn't do any conversions on properties that are
prefixed with the "%profile." prefixes. The fact that the profile prefix
is present means the property belongs to an inactive profile. The active
profile property names should be already normalized by the
ProfileConfigSourceInterceptor interceptor.

(cherry picked from commit bb148be6843b16a6655e0e96ff9641cdd080b6ed)

radcortez committed:
Update SmallRye Config to 2.6.1

(cherry picked from commit 51c4cb94cfc1270ea1f4918acae4275f18d88cae)

kdubb committed:
Pass checked exceptions unwrapped for Kotlin methods

(cherry picked from commit c524fe657a97991926a7fb9fafdda0f69078a39d)

Commit:
Add commons-logging exclusion to keycloak-admin extension

Commons-logging appears in classpath with a gradle setup.
So, I added an exclusion in order to prevent this lib
showing up whenever the extension is added.

The presence of apache commons-logging in classpath breaks
runtime execution with native apps.

(cherry picked from commit 1436100d189ef61e24036c80aa1daa78426203d4)

geoand committed:
Ensure that RESTEasy Reactive violation report doesn't break Hibernate Validator in native

Fixes: #21516
(cherry picked from commit 34648edf85a05a42e433afe6cd195bc1e6d865ff)

mswiderski committed:
fixed escaping of properties for azure-functions-http archetype

(cherry picked from commit b54fe759a3d5d2f4fe184ec3cf708a9bec1125db)

mkouba committed:
Qute - fix ChildResolutionContext.getExtendingBlock()

- it should always try the parent context as a fallback
- resolves #21584

(cherry picked from commit 6c99e47632cdc39dc77e4462d2487987b411bff1)

geoand committed:
Make quarkus-jdbc-mysql work properly in native mode for Java 17

Fixes: #21366
(cherry picked from commit c59ef65fafb1e04b68927ac677a7d429ad915c03)

loicmathieu committed:
App Engine Standard should always use the uberjar

Fixes #21564

(cherry picked from commit dda035968ea1fd504718caa1b135d8ec18997bf9)

geoand committed:
Ensure that Derby Dev Services works properly

Fixes: #21639
(cherry picked from commit adaa1b90a1afd003155d926b47808dcb5c069f59)

aloubyansky committed:
Propagate the Maven settings argument when testing codestart-generated projects

(cherry picked from commit 24946e6ef272bf705b28b6846c3714a8c2a2956d)

geoand committed:
Fix InputStream handling as Response entity in RESTEasy Classic

Fixes: #21602
(cherry picked from commit 350965b6f19b322efb38d7babac505b23ad39fab)

kdubb committed:
Test intercepted Kotlin methods do not wrap exceptions

(cherry picked from commit 3ccae3dbe02c278b8f220a2ee86ba42e48d44937)

mswiderski committed:
Fixed azure config function.json file to be properly handled by the azure functions http maven archetype

(cherry picked from commit 1d0be3d3d9a80962c30592bf3e0e701bb5477412)

glefloch committed:
Set produced Logger as DefaultBean

(cherry picked from commit f9e0f70281c9d1f4747ba44b5e08d81b13d4c306)

gsmet committed:
Remove dot from guide title

(cherry picked from commit fd7a32f0ec274d444ea07fb0229529a8cbed9e20)

Sanne committed:
Upgrade to MySQL JDBC driver 8.0.27

(cherry picked from commit 6ba4559a2f66dea96712474c9dfb6d2f9b5e5d09)

Sgitario committed:
Spring Data: Fix interface projection using Page and Slice

The interface project worked fine using Lists, but not using Page or Slice. Example:

```java
@Repository
public interface BasicTypeDataRepository extends CrudRepository<BasicTypeData, Integer> {

    // works
    @Query(value = "select id, doubleValue from BasicTypeData")
    List<WithDoubleValue> findAllDoubleValuesToList();

   // it does not work (1)
    @Query(value = "select id, doubleValue from BasicTypeData")
    Page<WithDoubleValue> findAllDoubleValuesToPage();
}
```

This PR fixes the interface projection when returning a Page or a Slice object (1).

Fixes https://github.com/quarkusio/quarkus/issues/21616

(cherry picked from commit a04b79793fb10da6b28b9126dd201078c70bb8b6)

Sanne committed:
Update docs/src/main/asciidoc/datasource.adoc

Co-authored-by: sberyozkin <sberyozkin@gmail.com>
(cherry picked from commit 6e4677f176605a7042072c6ef76578e65af1e365)

famod commented:
@gsmet those dev mode test failures look familiar, I think you'll need to include #21601.

Sgitario committed:
Resteasy Reactive - Avoid NPE when using runtime converters

The NPE happens here: https://github.com/quarkusio/quarkus/blob/e6ffbe43243f9c852c61aafc307515be0915c6c8/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ClassInjectorTransformer.java#L222.

However, it's expected that the delegated converter is null when there are runtime converts. See related lines: https://github.com/quarkusio/quarkus/blob/bd87966540990fa68ec3769e22e67df1854b08a5/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusServerEndpointIndexer.java#L169 and https://github.com/quarkusio/quarkus/blob/bd87966540990fa68ec3769e22e67df1854b08a5/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusServerEndpointIndexer.java#L173.

Therefore, as It's possible that the delegate is null when there are runtime converters, we check whether the delegate is null and if it is, we don't use it to avoid the NPE. The rest works as expected as proven in the test.

Fixes https://github.com/quarkusio/quarkus/issues/21664

(cherry picked from commit b66ba75b5552fdb2d8999a5697c8959d9a95cbf5)

gsmet commented:
@famod thanks for the heads up :)
